src/scanspec/core.py

```diff
@@ -483,13 +501,27 @@ def zip_gap(gaps: Sequence[GapArray]) -> GapArray:
             # gap[i] = self.gap[i] | other.gap[i]
             return np.logical_or.reduce(gaps)
 
-        return _merge_frames(self, other, dict_merge=zip_dict, gap_merge=zip_gap)
+        def zip_duration(
+            durations: Sequence[DurationArray | None],
+        ) -> DurationArray | None:
+            # assert len(durations) == 1, "Can't have more than one durations array"
+            return durations[-1]
```

Comment:
Need to check that only one duration array is given:

```suggestion
            # We catch the case of all None in merge_frames, so only handle multiple durations here
            specified_durations = [d for d in durations if d is not None]
            if len(specified_durations) != 1:
                raise ValueError("Can't have more than one durations array")
            return specified_durations[0]
```

Can we have a test for this too please